UnityProject/Assets/Scripts/Items/Others/Jetpack.cs

```diff
@@ -87,5 +82,13 @@ public void ServerPerformInteraction(InventoryApply interaction)
 			}
 			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, PushUpdate);
 		}
+
+		public static void PushPlayerInFacedDirection(PlayerScript playerScript, GasContainer gasContainer, float gasRelease = 10)
+		{
+			if (playerScript.objectPhysics.CanPush(playerScript.CurrentDirection.ToLocalVector2Int()) == false) return;
+			playerScript.objectPhysics.NewtonianPush(playerScript.CurrentDirection.ToLocalVector2Int(),1f);
```

Comment:
this needs to be replicated on client, Otherwise the  manual desynchronise Checks will have to fix the player's momentum

Reply:
why doesn't the movement script handle that

Reply:
because you would need to trigger the movement on the client before the server because it has to act on the client first, and then be validated on server and then act on server, 

not exactly something you can integrate on will interact

Reply:
but im not doing it on will interact and the way you're putting it it means that the server has no way of moving the player and that the client has to be the one that issues that command which means that people can use this to cheat.

Reply:
review pls